### PR TITLE
[EXPRESSIONS] Add log with custom base

### PR DESCRIPTION
**Pull request by jhasm**
### Resolves 
1. https://github.com/Eventual-Inc/Daft/issues/2315

### Summary
This change adds a new expression `log` to compute log with custom base.

### Testing
* Unit tests added to cover the new changes.

### Related PRs
* https://github.com/Eventual-Inc/Daft/pull/2168 used as a reference to keep all log related expressions uniform

### CODE REVIEW

**Review by colin-ho — COMMENTED** [review]
Thanks @jhasm ! Just one comment

File: src/daft-dsl/src/functions/numeric/mod.rs [hunk 1]
```diff
@@ -16,13 +16,35 @@ use round::RoundEvaluator;
...
+}
+
+impl Eq for F64Wrapper {}
+impl Hash for F64Wrapper {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        // Use the to_bits method to convert the f64 into its raw bytes representation
+        self.0.to_bits().hash(state);
+    }
+}
+
```
[the remark this answers is not in the document]
Comment (jhasm) [reply src/daft-dsl/src/functions/numeric/mod.rs:47]: Ah cool, thanks! I didn't know that existed and was surely hoping someone will suggest a better alternative to my solution. 
Let me change this.

[the remark this answers is not in the document]
Comment (jhasm) [reply src/daft-dsl/src/functions/numeric/mod.rs]: Updated the code to use `FloatWrapper`

**Review by colin-ho — APPROVED** [review]
LGTM, thanks @jhasm !

**Pull request merged**